client/src/schema/selection.ts

```diff
@@ -19,7 +20,8 @@ type OmitMethods<T> = {
   [key in keyof T as T[key] extends Function ? never : key]: T[key];
 };
 
-export type Selectable<T> = Omit<OmitQueries<OmitMethods<T>>, 'id' | 'xata'>;
+type InternalProperties = 'xata' | 'read' | 'update' | 'delete';
```

Comment:
This list may change. We know `xata` is always going to be there, but we may add more functions in the future. I had a type to omit functions, what do you think about using that?

Reply:
I have a better solution in mind:

```suggestion
type InternalProperties = keyof Omit<XataRecord, 'id'>;
```